Allow changing weight in `TextElement` and similar skin elements

frenzibyte opened the pull request:
![CleanShot 2025-06-26 at 07 19 23](https://github.com/user-attachments/assets/38626dea-d424-497a-8474-542ad6751743)

Applicable to `TextElement`, `PlayerName`, and `BeatmapAttributeText`.

---

- Addresses https://github.com/ppy/osu/discussions/33864

Seemed like a nice feature to have while also being easy to implement.

frenzibyte committed:
Support changing weight in text-based elements

peppy commented on osu.Game/Graphics/OsuFont.cs:
```diff
@@ -177,31 +179,37 @@ public enum FontWeight
         /// <summary>
         /// Equivalent to weight 300.
         /// </summary>
+        [LocalisableDescription(typeof(FontStrings), nameof(FontStrings.Light))]
```
I don't know how much value there is in localising these..

bdach replied:
I would say there is none.

Qunas101 replied:
Yeah, I agree with that. I've seen bald and semi-bold translated to my language, but everything else just sounds wacky

bdach commented on osu.Game/Localisation/SkinComponents/SkinnableComponentStrings.cs:
```diff
@@ -79,6 +79,16 @@ public static class SkinnableComponentStrings
+        /// </summary>
+        public static LocalisableString TextWeight => new TranslatableString(getKey(@"text_weight"), @"Text weight");
+
+        /// <summary>
+        /// "The weight of the text."
+        /// </summary>
+        public static LocalisableString TextWeightDescription => new TranslatableString(getKey(@"text_weight_description"), @"The weight of the text.");
```
would get rid of this as well because this is zero new information over the label

frenzibyte committed:
Remove font weight localisation

peppy committed:
Remove pointless tooltip

I agree that these are pointless and we should probably remove the
others and stop wasting localiser's time on these.
